charts/catalog/templates/controller-manager-deployment.yaml

```diff
@@ -54,6 +54,7 @@ spec:
         - --service-catalog-api-server-url
         - https://{{ template "fullname" . }}-apiserver
         {{- end }}
+        - "--service-catalog-insecure-skip-verify=true"
```

Comment:
I might be wrong, but if we want to disable this flag, we also need to pass some extra information in the controller-manager (information about cerificate and authority we trust to). Without it the TLS verification will still fail.
In other words, this flag doesn't fix the original problem, just removes the hack from the code.

I haven't fully analyzed the part of interaction between controller-manager and service-catalog, and would prefer to ask someone more familiar with it (you? 😉) to test it with disabled flag and raise a separate PR adressing this.